tests/OrganizationsMetadataTest.php

```diff
@@ -71,46 +72,64 @@ public function allProjectRoles(): array
         ];
     }
 
+    public function providers(): array
+    {
+        return [
+            'system provider' => [
+                self::PROVIDER_SYSTEM,
+            ],
+            'user provider' => [
+                self::PROVIDER_USER,
+            ],
+        ];
+    }
+
     public function testNormalUserCanNotManageMetadata(): void
     {
         $this->client->addUserToOrganization($this->organization['id'], ['email' => $this->superAdmin['email']]);
         $organizationId = $this->organization['id'];
 
         $this->createUserMetadata($this->client, $organizationId);
 
-        $this->cannotManageMetadata($this->normalUserClient);
+        $this->cannotManageUserMetadata($this->normalUserClient);
+        $this->cannotManageSystemMetadata($this->normalUserClient);
     }
 
-    public function testSuperAdminCanManageMetadata(): void
+    /**
+     * @dataProvider providers
+     */
+    public function testSuperAdminCanManageMetadata(string $provider): void
     {
         $this->client->addUserToOrganization($this->organization['id'], ['email' => $this->normalUser['email']]);
 
-        $this->createUserMetadata($this->client, $this->organization['id']);
+        $this->createMetadata($this->client, $this->organization['id'], $provider);
 
         // trying for allowAutoJoin=false. Metadata should be just updateded
         $this->normalUserClient->updateOrganization($this->organization['id'], [
             'allowAutoJoin' => false,
         ]);
-        $userMetadata = $this->createUserMetadata($this->client, $this->organization['id']);
+        $userMetadata = $this->createMetadata($this->client, $this->organization['id'], $provider);
 
         $this->assertCount(2, $userMetadata);
-        $this->validateMetadataEquality(self::TEST_METADATA[1], $userMetadata[0], self::PROVIDER_USER);
-        $this->validateMetadataEquality(self::TEST_METADATA[0], $userMetadata[1], self::PROVIDER_USER);
+        $this->validateMetadataEquality(self::TEST_METADATA[1], $userMetadata[0], $provider);
+        $this->validateMetadataEquality(self::TEST_METADATA[0], $userMetadata[1], $provider);
 
         $metadataArray = $this->client->listOrganizationMetadata($this->organization['id']);
 
         $this->assertCount(2, $metadataArray);
-        $this->validateMetadataEquality(self::TEST_METADATA[1], $metadataArray[0], self::PROVIDER_USER);
-        $this->validateMetadataEquality(self::TEST_METADATA[0], $metadataArray[1], self::PROVIDER_USER);
+        $this->validateMetadataEquality(self::TEST_METADATA[1], $metadataArray[0], $provider);
+        $this->validateMetadataEquality(self::TEST_METADATA[0], $metadataArray[1], $provider);
     }
 
-    public function testSuperAdminWithoutMFACannotManageUserMetadata(): void
+
+    public function testSuperAdminWithoutMFACannotManageMetadata(): void
```

Comment:
test by mal testovat aj to ze nejde nastavi system metadata, testujes len to ze nejdu user metadata

Reply:
shit, mel jsem to tak, ale pak jsem v tom delal jeste nejake upravy a zrusil jsem to. vyresil jsem to tedy https://github.com/keboola/kbc-manage-api-php-client/pull/131/commits/e976eb52f3f2e9ed02f4daaca18ba6d7150748c6